src/applications/detection/main.cpp

```diff
@@ -51,236 +52,353 @@ using boost::timer;
+              << "The input file \"" << cmdline._filename << "\" is missing" << std::endl;
       return EXIT_FAILURE;
     }
-  }else{
+  }
+  else
+  {
     std::cerr << std::endl
-        << "An input file is required" << std::endl;
-    cmdline.usage( argv[0] );
+            << "An input file is required" << std::endl;
+    cmdline.usage(argv[0]);
     return EXIT_FAILURE;
   }
 
 #ifdef WITH_CUDA
-  popart::pop_cuda_only_sync_calls( cmdline._switchSync );
+  popart::pop_cuda_only_sync_calls(cmdline._switchSync);
 #endif
 
   // Check the (optional) parameters path
-  std::size_t nCrowns = std::atoi(cmdline._nCrowns.c_str());
+  const std::size_t nCrowns = std::atoi(cmdline._nCrowns.c_str());
   cctag::Parameters params(nCrowns);
-  
-  if( cmdline._paramsFilename != "" ) {
-    if (!boost::filesystem::exists( cmdline._paramsFilename )) {
+
+  if(!cmdline._paramsFilename.empty())
+  {
+    if(!boost::filesystem::exists(cmdline._paramsFilename))
+    {
       std::cerr << std::endl
-        << "The input file \""<< cmdline._paramsFilename << "\" is missing" << std::endl;
+              << "The input file \"" << cmdline._paramsFilename << "\" is missing" << std::endl;
       return EXIT_FAILURE;
     }
 
     // Read the parameter file provided by the user
-    std::ifstream ifs( cmdline._paramsFilename.c_str() );
+    std::ifstream ifs(cmdline._paramsFilename);
     boost::archive::xml_iarchive ia(ifs);
     ia >> boost::serialization::make_nvp("CCTagsParams", params);
     CCTAG_COUT(params._nCrowns);
     CCTAG_COUT(nCrowns);
-    assert( nCrowns == params._nCrowns );
-  } else {
+    assert(nCrowns == params._nCrowns);
+  }
+  else
+  {
     // Use the default parameters and save them in defaultParameters.xml
     cmdline._paramsFilename = "defaultParameters.xml";
-    std::ofstream ofs( cmdline._paramsFilename.c_str() );
+    std::ofstream ofs(cmdline._paramsFilename);
     boost::archive::xml_oarchive oa(ofs);
     oa << boost::serialization::make_nvp("CCTagsParams", params);
     CCTAG_COUT("Parameter file not provided. Default parameters are used.");
   }
-  
+
   CCTagMarkersBank bank(params._nCrowns);
-  if ( !cmdline._cctagBankFilename.empty())
+  if(!cmdline._cctagBankFilename.empty())
   {
     bank = CCTagMarkersBank(cmdline._cctagBankFilename);
   }
 
 #ifdef WITH_CUDA
-  if( cmdline._useCuda ) {
-    params.setUseCuda( true );
-  } else {
-    params.setUseCuda( false );
+  if(cmdline._useCuda)
+  {
+    params.setUseCuda(true);
+  }
+  else
+  {
+    params.setUseCuda(false);
   }
 
-  if( cmdline._debugDir != "" ) {
-    params.setDebugDir( cmdline._debugDir );
+  if(cmdline._debugDir != "")
+  {
+    params.setDebugDir(cmdline._debugDir);
   }
 
-  popart::device_prop_t deviceInfo( false );
+  popart::device_prop_t deviceInfo(false);
 #endif // WITH_CUDA
 
-  bfs::path myPath( cmdline._filename );
+  bfs::path myPath(bfs::absolute(cmdline._filename));
```

Comment:
`bfs` ?! Be consistent, either use `bfs` or `boost::filesystem` everywhere, but don't mix them.